account.rb

```diff
@@ -0,0 +1,44 @@
+
+class Account
+
+    attr_reader :name, :currency, :balance, :nature
+
+    def initialize()
+        @name = ""
+        @currency = ""
+        @balance = 0
+        @nature = ""
+    end
+
+    def set_name(name)
```

Comment:
https://github.com/bbatsov/ruby-style-guide   
https://blog.metova.com/a-beginners-guide-to-ruby-getters-and-setters/  
read this, refactor your code, ping me